stl/inc/xhash

```diff
@@ -279,7 +285,7 @@ struct _Hash_vec {
             // nothrow hereafter
             if (_Oldsize != 0) {
                 _Destroy_range(_Mypair._Myval2._Myfirst, _Mypair._Myval2._Mylast);
-                _Alvec.deallocate(_Mypair._Myval2._Myfirst, _Oldsize);
+                _Alvec.deallocate(_Mypair._Myval2._Myfirst, capacity());
             }

```

Comment:
Could the old code ever get into a state where the old size was zero (so this conditional would be skipped), but the old capacity was nonzero (so we have allocated memory that would be leaked here)?

I *believe* the answer is no, due to the "min buckets" logic, in which case this is fine as-is.

Reply:
I suspect the answer is "no" as well, but without digging out the old code and performing a full analysis (including exceptions states) I'd rather defend against it. It's simple enough to do so.